src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -238,6 +238,45 @@ impl ScalarFunc {
         let week = t.get_time().iso_week().week();
         Ok(Some(i64::from(week)))
     }
+
+    #[inline]
+    pub fn year_week_with_mode(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let t: Cow<Time> = try_opt!(self.children[0].eval_time(ctx, row));
+        if t.is_zero() {
+            return handle_incorrect_datetime_error(ctx, t).map(|_| None);
+        }
+        let mode = match self.children[1].eval_int(ctx, row) {
+            Err(e) => return Err(e),
+            Ok(None) => 0,
+            Ok(Some(num)) => num,
+        };
+        let (year, week) = t
+            .get_time()
+            .year_week(WeekMode::from_bits_truncate(mode as u32));
+        let mut result = i64::from(week + year * 100);
+        if result < 0 {
+            result = i64::from(u32::max_value());
+        }
+        Ok(Some(result))
+    }
+
+    #[inline]
+    pub fn year_week_without_mode(
+        &self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<i64>> {
+        let t: Cow<Time> = try_opt!(self.children[0].eval_time(ctx, row));
+        if t.is_zero() {
```

Comment:
TiDB uses [InvalidZero](https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L1496) here, although I'm not sure what's the difference. @breeswish do you have any idea about this?

Reply:
@breeswish PTAL

Reply:
We have "invalid_zero" function in TiKV and it should be used.

Reply:
@breeswish I don't know why yearWeekWithMode use isZero, but yearWeekWithoutMode use invalid_zero in TiDB.  It is wired.

Reply:
It should be a mistake :( 🤣 Please use `invalid_zero`. Thanks!